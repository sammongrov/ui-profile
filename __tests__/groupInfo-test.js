import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import GroupInfo from '../GroupInfo';
import DbManager from '../../app/DBManager';

configure({ adapter: new Adapter() });
jest.mock('react-native-router-flux', () => ({
  Actions: {
    pop: jest.fn(),
    currentScene: 'GroupInfoScene',
    MemberInfo: jest.fn(),
  },
}));

jest.mock('../../app/DBManager', () => {
  const dbManager = {
    app: {},
    user: {},
    group: {},
    _taskManager: {},
  };
  return dbManager;
});

const groupInfo = {
  _id: '7X856YIks9',
  name: 'NewUser',
  title: 'NewUser',
  type: 'd',
  avatar: 'newUser',
  unread: 0,
  status: 'busy',
};

const users = {
  '0': { _id: 'XNN1986P1', username: 'NewUser', name: 'Four', status: 'away' },
  '1': {
    _id: 'XPP89611P',
    username: 'divergent',
    name: 'Trix',
    status: 'online',
  },
};

DbManager.group.findById = jest.fn(() => groupInfo);
DbManager._taskManager.chat = { getGroupUsers: jest.fn(() => users) };
DbManager.app.app = { host: 'test.mongrov.com' };

it('GroupInfo renders correctly with props', () => {
  const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('GroupInfo unmounts correctly', () => {
  const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
  const treeInstance = tree.getInstance();
  tree.unmount();
  expect(treeInstance._isMounted).toBe(false);
});

describe('getOnlineMembers method is called', () => {
  it('with the users object', () => {
    const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.getInstance();
    const result = treeInstance.getOnlineMembers(users);
    expect(result).toMatch('1');
  });

  it('with not an object', () => {
    const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.getInstance();
    const result = treeInstance.getOnlineMembers('members');
    expect(result).toMatch('0');
  });
});

describe('fetchGroupMembers method is called', () => {
  it('with isMounted equals true', async () => {
    const tree = shallow(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.instance();
    treeInstance._isMounted = true;
    treeInstance.createMembersList = jest.fn(() => users);
    expect.assertions(4);
    await treeInstance.fetchGroupMembers();
    expect(DbManager._taskManager.chat.getGroupUsers).toBeCalled();
    expect(treeInstance.createMembersList).toBeCalled();
    expect(treeInstance._isMounted).toBe(true);
    expect(tree.state().members).toEqual(users);
  });

  it('with isMounted equals true', async () => {
    DbManager._taskManager.chat.getGroupUsers.mockClear();
    const tree = shallow(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.instance();
    treeInstance._isMounted = false;
    treeInstance.createMembersList = jest.fn(() => users);
    expect.assertions(4);
    await treeInstance.fetchGroupMembers();
    expect(DbManager._taskManager.chat.getGroupUsers).toBeCalled();
    expect(treeInstance.createMembersList).not.toBeCalled();
    expect(treeInstance._isMounted).toBe(false);
    expect(tree.state().members).toEqual([]);
  });
});

describe('userprofile method is called', () => {
  it('a current scene is "groupInfoScene"', () => {
    const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.getInstance();
    treeInstance.userprofile('XPP89611P');
    expect(Actions.MemberInfo).toBeCalled();
  });

  it('with not an object', () => {
    Actions.MemberInfo.mockClear();
    Actions.currentScene = 'MemberInfoScene';
    const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.getInstance();
    treeInstance.userprofile('XPP89611P');
    expect(Actions.MemberInfo).not.toBeCalled();
  });
});

describe('createMembersList method is called', () => {
  it('with the users object', () => {
    const groupUsers = [
      { _id: 'XNN1986P1', status: 'away', avatarURL: 'https://four.avatar.io' },
      { _id: 'XPP89611P', username: 'divergent', status: 'online' },
    ];
    const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.getInstance();
    const members = [
      {
        id: 'XNN1986P1',
        title: '',
        status: 'away',
        subTitle: '@undefined',
        avatar_url: 'https://four.avatar.io',
      },
      {
        id: 'XPP89611P',
        title: 'divergent',
        status: 'online',
        subTitle: '@divergent',
        avatar_url: undefined,
      },
    ];
    const result = treeInstance.createMembersList(groupUsers);
    expect(result[0]).toMatchObject(members[0]);
    expect(result[1]).toMatchObject(members[1]);
  });

  it('with not an object', () => {
    const tree = renderer.create(<GroupInfo memberId="7X856YIks9" />);
    const treeInstance = tree.getInstance();
    const result = treeInstance.createMembersList('members');
    expect(result).toEqual([]);
  });
});

it('GroupInfo renders with a description and an announcement', () => {
  const tree = shallow(<GroupInfo memberId="7X856YIks9" />);
  const groupObj = {
    _id: '25X856YIks9',
    // name: 'First jumper',
    description: 'Brave first jumper',
    announcement: 'Let us jump',
    type: 'private',
    avatarURL: 'first-jumper.net',
    unread: 0,
  };
  tree.setState({ groupObj });
  const profile = tree
    .find('Profile')
    .first()
    .props();
  expect(profile.profileTitle).toMatch('');
  expect(profile.profileUname).toMatch(groupObj.description);
  expect(profile.profileAnnouncement).toMatch(groupObj.announcement);
});

// onPress events
it('Profile onPress', () => {
  const tree = shallow(<GroupInfo memberId="7X856YIks9" />);
  const profile = tree.find('Profile').first();
  profile.props().onPress();
  expect(profile.props().onPress).toBeInstanceOf(Function);
});

it('ProfileList onPress', async () => {
  Actions.currentScene = 'GroupInfoScene';
  Actions.MemberInfo.mockClear();
  const tree = shallow(<GroupInfo memberId="7X856YIks9" />);
  const treeInstance = tree.instance();
  treeInstance._isMounted = true;
  await treeInstance.fetchGroupMembers();
  tree.update();
  const profileList = tree.find('ProfileList');
  const flatList = profileList
    .shallow()
    .find('FlatList')
    .shallow();
  const data = flatList.props().data[0];
  const listItem = shallow(flatList.props().renderItem({ item: data }));
  listItem.props().onPress();
  expect.assertions(1);
  expect(Actions.MemberInfo).toHaveBeenCalledTimes(1);
});

it('ProfileList onAvatarPress', async () => {
  Actions.currentScene = 'GroupInfoScene';
  Actions.MemberInfo.mockClear();
  const tree = shallow(<GroupInfo memberId="7X856YIks9" />);
  const treeInstance = tree.instance();
  treeInstance._isMounted = true;
  await treeInstance.fetchGroupMembers();
  tree.update();
  const profileList = tree.find('ProfileList');
  const flatList = profileList
    .shallow()
    .find('FlatList')
    .shallow();
  const data = flatList.props().data[0];
  const listItem = shallow(flatList.props().renderItem({ item: data }));
  const avatar = listItem.find('Avatar');
  avatar.props().onAvatarPress();
  expect.assertions(1);
  expect(Actions.MemberInfo).toHaveBeenCalledTimes(1);
});
