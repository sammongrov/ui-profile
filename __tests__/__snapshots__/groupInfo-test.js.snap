// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupInfo renders correctly with props 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="material-community"
            >
              ÔÖÅ
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Group Info
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <RCTScrollView
        showsVerticalScrollIndicator={false}
      >
        <View>
          <View
            style={
              Array [
                undefined,
                undefined,
                Object {
                  "paddingBottom": 5,
                },
              ]
            }
          >
            <View>
              <View
                style={
                  Array [
                    undefined,
                    Object {
                      "position": "relative",
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#d134d1",
                      "borderRadius": 40,
                      "height": 80,
                      "opacity": 1,
                      "width": 80,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "fontFamily": "OpenSans-Regular",
                        },
                        null,
                      ]
                    }
                  >
                    N
                  </Text>
                </View>
              </View>
            </View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "OpenSans-Regular",
                  },
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              NewUser
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontFamily": "OpenSans-Regular",
                  },
                  Array [
                    undefined,
                    undefined,
                  ],
                ]
              }
            >
              0 members (0) online
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "justifyContent": "center",
                "paddingBottom": 20,
                "paddingTop": 10,
              }
            }
          >
            <View>
              <RNSVGSvgView
                bbHeight="120"
                bbWidth="120"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                    false,
                    Object {
                      "flex": 0,
                      "height": 120,
                      "width": 120,
                    },
                  ]
                }
              >
                <RNSVGRect
                  fill={
                    Array [
                      0,
                      1,
                      1,
                      1,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  height="120"
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "fill",
                    ]
                  }
                  rotation={0}
                  rx="0"
                  ry="0"
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="1"
                  width="120"
                  x="0"
                  y="0"
                />
                <RNSVGPath
                  d="M0 2.0689655172413794 L28.965517241379313 2.0689655172413794 M33.10344827586207 2.0689655172413794 L49.65517241379311 2.0689655172413794 M53.793103448275865 2.0689655172413794 L78.62068965517241 2.0689655172413794 M91.0344827586207 2.0689655172413794 L120.00000000000001 2.0689655172413794 M0 6.206896551724139 L4.137931034482759 6.206896551724139 M24.827586206896555 6.206896551724139 L28.965517241379313 6.206896551724139 M37.24137931034483 6.206896551724139 L57.931034482758626 6.206896551724139 M66.20689655172414 6.206896551724139 L74.48275862068967 6.206896551724139 M78.62068965517241 6.206896551724139 L82.75862068965517 6.206896551724139 M91.0344827586207 6.206896551724139 L95.17241379310346 6.206896551724139 M115.86206896551725 6.206896551724139 L120.00000000000001 6.206896551724139 M0 10.344827586206897 L4.137931034482759 10.344827586206897 M8.275862068965518 10.344827586206897 L20.689655172413794 10.344827586206897 M24.827586206896555 10.344827586206897 L28.965517241379313 10.344827586206897 M37.24137931034483 10.344827586206897 L41.37931034482759 10.344827586206897 M45.51724137931035 10.344827586206897 L53.793103448275865 10.344827586206897 M57.931034482758626 10.344827586206897 L62.06896551724138 10.344827586206897 M66.20689655172414 10.344827586206897 L82.75862068965517 10.344827586206897 M91.0344827586207 10.344827586206897 L95.17241379310346 10.344827586206897 M99.31034482758622 10.344827586206897 L111.72413793103449 10.344827586206897 M115.86206896551725 10.344827586206897 L120.00000000000001 10.344827586206897 M0 14.482758620689657 L4.137931034482759 14.482758620689657 M8.275862068965518 14.482758620689657 L20.689655172413794 14.482758620689657 M24.827586206896555 14.482758620689657 L28.965517241379313 14.482758620689657 M33.10344827586207 14.482758620689657 L37.24137931034483 14.482758620689657 M53.793103448275865 14.482758620689657 L66.20689655172414 14.482758620689657 M74.48275862068967 14.482758620689657 L78.62068965517241 14.482758620689657 M91.0344827586207 14.482758620689657 L95.17241379310346 14.482758620689657 M99.31034482758622 14.482758620689657 L111.72413793103449 14.482758620689657 M115.86206896551725 14.482758620689657 L120.00000000000001 14.482758620689657 M0 18.620689655172416 L4.137931034482759 18.620689655172416 M8.275862068965518 18.620689655172416 L20.689655172413794 18.620689655172416 M24.827586206896555 18.620689655172416 L28.965517241379313 18.620689655172416 M33.10344827586207 18.620689655172416 L57.931034482758626 18.620689655172416 M62.06896551724138 18.620689655172416 L78.62068965517241 18.620689655172416 M91.0344827586207 18.620689655172416 L95.17241379310346 18.620689655172416 M99.31034482758622 18.620689655172416 L111.72413793103449 18.620689655172416 M115.86206896551725 18.620689655172416 L120.00000000000001 18.620689655172416 M0 22.758620689655174 L4.137931034482759 22.758620689655174 M24.827586206896555 22.758620689655174 L28.965517241379313 22.758620689655174 M33.10344827586207 22.758620689655174 L41.37931034482759 22.758620689655174 M57.931034482758626 22.758620689655174 L62.06896551724138 22.758620689655174 M66.20689655172414 22.758620689655174 L70.3448275862069 22.758620689655174 M78.62068965517241 22.758620689655174 L86.89655172413794 22.758620689655174 M91.0344827586207 22.758620689655174 L95.17241379310346 22.758620689655174 M115.86206896551725 22.758620689655174 L120.00000000000001 22.758620689655174 M0 26.896551724137936 L28.965517241379313 26.896551724137936 M33.10344827586207 26.896551724137936 L37.24137931034483 26.896551724137936 M41.37931034482759 26.896551724137936 L45.51724137931035 26.896551724137936 M49.65517241379311 26.896551724137936 L53.793103448275865 26.896551724137936 M57.931034482758626 26.896551724137936 L62.06896551724138 26.896551724137936 M66.20689655172414 26.896551724137936 L70.3448275862069 26.896551724137936 M74.48275862068967 26.896551724137936 L78.62068965517241 26.896551724137936 M82.75862068965517 26.896551724137936 L86.89655172413794 26.896551724137936 M91.0344827586207 26.896551724137936 L120.00000000000001 26.896551724137936 M33.10344827586207 31.034482758620694 L41.37931034482759 31.034482758620694 M49.65517241379311 31.034482758620694 L53.793103448275865 31.034482758620694 M62.06896551724138 31.034482758620694 L66.20689655172414 31.034482758620694 M70.3448275862069 31.034482758620694 L86.89655172413794 31.034482758620694 M0 35.17241379310345 L4.137931034482759 35.17241379310345 M16.551724137931036 35.17241379310345 L20.689655172413794 35.17241379310345 M24.827586206896555 35.17241379310345 L37.24137931034483 35.17241379310345 M41.37931034482759 35.17241379310345 L49.65517241379311 35.17241379310345 M53.793103448275865 35.17241379310345 L62.06896551724138 35.17241379310345 M66.20689655172414 35.17241379310345 L78.62068965517241 35.17241379310345 M82.75862068965517 35.17241379310345 L107.58620689655173 35.17241379310345 M115.86206896551725 35.17241379310345 L120.00000000000001 35.17241379310345 M4.137931034482759 39.31034482758621 L8.275862068965518 39.31034482758621 M20.689655172413794 39.31034482758621 L24.827586206896555 39.31034482758621 M33.10344827586207 39.31034482758621 L49.65517241379311 39.31034482758621 M57.931034482758626 39.31034482758621 L62.06896551724138 39.31034482758621 M66.20689655172414 39.31034482758621 L78.62068965517241 39.31034482758621 M82.75862068965517 39.31034482758621 L86.89655172413794 39.31034482758621 M91.0344827586207 39.31034482758621 L95.17241379310346 39.31034482758621 M99.31034482758622 39.31034482758621 L120.00000000000001 39.31034482758621 M0 43.44827586206897 L4.137931034482759 43.44827586206897 M8.275862068965518 43.44827586206897 L12.413793103448278 43.44827586206897 M16.551724137931036 43.44827586206897 L20.689655172413794 43.44827586206897 M24.827586206896555 43.44827586206897 L37.24137931034483 43.44827586206897 M45.51724137931035 43.44827586206897 L66.20689655172414 43.44827586206897 M74.48275862068967 43.44827586206897 L78.62068965517241 43.44827586206897 M95.17241379310346 43.44827586206897 L99.31034482758622 43.44827586206897 M115.86206896551725 43.44827586206897 L120.00000000000001 43.44827586206897 M0 47.58620689655173 L4.137931034482759 47.58620689655173 M12.413793103448278 47.58620689655173 L20.689655172413794 47.58620689655173 M28.965517241379313 47.58620689655173 L37.24137931034483 47.58620689655173 M45.51724137931035 47.58620689655173 L49.65517241379311 47.58620689655173 M53.793103448275865 47.58620689655173 L57.931034482758626 47.58620689655173 M70.3448275862069 47.58620689655173 L91.0344827586207 47.58620689655173 M103.44827586206897 47.58620689655173 L107.58620689655173 47.58620689655173 M111.72413793103449 47.58620689655173 L120.00000000000001 47.58620689655173 M0 51.72413793103449 L16.551724137931036 51.72413793103449 M24.827586206896555 51.72413793103449 L28.965517241379313 51.72413793103449 M41.37931034482759 51.72413793103449 L45.51724137931035 51.72413793103449 M49.65517241379311 51.72413793103449 L57.931034482758626 51.72413793103449 M66.20689655172414 51.72413793103449 L74.48275862068967 51.72413793103449 M78.62068965517241 51.72413793103449 L82.75862068965517 51.72413793103449 M86.89655172413794 51.72413793103449 L91.0344827586207 51.72413793103449 M95.17241379310346 51.72413793103449 L99.31034482758622 51.72413793103449 M111.72413793103449 51.72413793103449 L115.86206896551725 51.72413793103449 M20.689655172413794 55.862068965517246 L24.827586206896555 55.862068965517246 M33.10344827586207 55.862068965517246 L49.65517241379311 55.862068965517246 M53.793103448275865 55.862068965517246 L70.3448275862069 55.862068965517246 M74.48275862068967 55.862068965517246 L78.62068965517241 55.862068965517246 M91.0344827586207 55.862068965517246 L120.00000000000001 55.862068965517246 M0 60.00000000000001 L8.275862068965518 60.00000000000001 M16.551724137931036 60.00000000000001 L20.689655172413794 60.00000000000001 M24.827586206896555 60.00000000000001 L37.24137931034483 60.00000000000001 M62.06896551724138 60.00000000000001 L66.20689655172414 60.00000000000001 M74.48275862068967 60.00000000000001 L78.62068965517241 60.00000000000001 M82.75862068965517 60.00000000000001 L86.89655172413794 60.00000000000001 M99.31034482758622 60.00000000000001 L111.72413793103449 60.00000000000001 M115.86206896551725 60.00000000000001 L120.00000000000001 60.00000000000001 M0 64.13793103448276 L8.275862068965518 64.13793103448276 M12.413793103448278 64.13793103448276 L16.551724137931036 64.13793103448276 M28.965517241379313 64.13793103448276 L33.10344827586207 64.13793103448276 M37.24137931034483 64.13793103448276 L41.37931034482759 64.13793103448276 M45.51724137931035 64.13793103448276 L53.793103448275865 64.13793103448276 M57.931034482758626 64.13793103448276 L66.20689655172414 64.13793103448276 M70.3448275862069 64.13793103448276 L86.89655172413794 64.13793103448276 M91.0344827586207 64.13793103448276 L99.31034482758622 64.13793103448276 M111.72413793103449 64.13793103448276 L120.00000000000001 64.13793103448276 M0 68.27586206896552 L8.275862068965518 68.27586206896552 M12.413793103448278 68.27586206896552 L16.551724137931036 68.27586206896552 M20.689655172413794 68.27586206896552 L28.965517241379313 68.27586206896552 M33.10344827586207 68.27586206896552 L37.24137931034483 68.27586206896552 M41.37931034482759 68.27586206896552 L45.51724137931035 68.27586206896552 M49.65517241379311 68.27586206896552 L62.06896551724138 68.27586206896552 M70.3448275862069 68.27586206896552 L74.48275862068967 68.27586206896552 M82.75862068965517 68.27586206896552 L91.0344827586207 68.27586206896552 M95.17241379310346 68.27586206896552 L99.31034482758622 68.27586206896552 M111.72413793103449 68.27586206896552 L115.86206896551725 68.27586206896552 M0 72.41379310344828 L12.413793103448278 72.41379310344828 M16.551724137931036 72.41379310344828 L20.689655172413794 72.41379310344828 M28.965517241379313 72.41379310344828 L37.24137931034483 72.41379310344828 M41.37931034482759 72.41379310344828 L45.51724137931035 72.41379310344828 M49.65517241379311 72.41379310344828 L53.793103448275865 72.41379310344828 M62.06896551724138 72.41379310344828 L70.3448275862069 72.41379310344828 M91.0344827586207 72.41379310344828 L107.58620689655173 72.41379310344828 M111.72413793103449 72.41379310344828 L120.00000000000001 72.41379310344828 M12.413793103448278 76.55172413793105 L16.551724137931036 76.55172413793105 M20.689655172413794 76.55172413793105 L41.37931034482759 76.55172413793105 M49.65517241379311 76.55172413793105 L70.3448275862069 76.55172413793105 M107.58620689655173 76.55172413793105 L111.72413793103449 76.55172413793105 M115.86206896551725 76.55172413793105 L120.00000000000001 76.55172413793105 M12.413793103448278 80.6896551724138 L16.551724137931036 80.6896551724138 M20.689655172413794 80.6896551724138 L24.827586206896555 80.6896551724138 M28.965517241379313 80.6896551724138 L37.24137931034483 80.6896551724138 M41.37931034482759 80.6896551724138 L49.65517241379311 80.6896551724138 M53.793103448275865 80.6896551724138 L57.931034482758626 80.6896551724138 M70.3448275862069 80.6896551724138 L74.48275862068967 80.6896551724138 M78.62068965517241 80.6896551724138 L82.75862068965517 80.6896551724138 M111.72413793103449 80.6896551724138 L120.00000000000001 80.6896551724138 M0 84.82758620689656 L16.551724137931036 84.82758620689656 M24.827586206896555 84.82758620689656 L41.37931034482759 84.82758620689656 M45.51724137931035 84.82758620689656 L57.931034482758626 84.82758620689656 M62.06896551724138 84.82758620689656 L66.20689655172414 84.82758620689656 M70.3448275862069 84.82758620689656 L74.48275862068967 84.82758620689656 M78.62068965517241 84.82758620689656 L107.58620689655173 84.82758620689656 M115.86206896551725 84.82758620689656 L120.00000000000001 84.82758620689656 M33.10344827586207 88.96551724137932 L41.37931034482759 88.96551724137932 M45.51724137931035 88.96551724137932 L49.65517241379311 88.96551724137932 M53.793103448275865 88.96551724137932 L62.06896551724138 88.96551724137932 M66.20689655172414 88.96551724137932 L78.62068965517241 88.96551724137932 M82.75862068965517 88.96551724137932 L86.89655172413794 88.96551724137932 M99.31034482758622 88.96551724137932 L103.44827586206897 88.96551724137932 M115.86206896551725 88.96551724137932 L120.00000000000001 88.96551724137932 M0 93.10344827586208 L28.965517241379313 93.10344827586208 M33.10344827586207 93.10344827586208 L37.24137931034483 93.10344827586208 M41.37931034482759 93.10344827586208 L45.51724137931035 93.10344827586208 M49.65517241379311 93.10344827586208 L53.793103448275865 93.10344827586208 M62.06896551724138 93.10344827586208 L66.20689655172414 93.10344827586208 M74.48275862068967 93.10344827586208 L86.89655172413794 93.10344827586208 M91.0344827586207 93.10344827586208 L95.17241379310346 93.10344827586208 M99.31034482758622 93.10344827586208 L111.72413793103449 93.10344827586208 M115.86206896551725 93.10344827586208 L120.00000000000001 93.10344827586208 M0 97.24137931034484 L4.137931034482759 97.24137931034484 M24.827586206896555 97.24137931034484 L28.965517241379313 97.24137931034484 M37.24137931034483 97.24137931034484 L41.37931034482759 97.24137931034484 M57.931034482758626 97.24137931034484 L62.06896551724138 97.24137931034484 M70.3448275862069 97.24137931034484 L86.89655172413794 97.24137931034484 M99.31034482758622 97.24137931034484 L103.44827586206897 97.24137931034484 M111.72413793103449 97.24137931034484 L115.86206896551725 97.24137931034484 M0 101.3793103448276 L4.137931034482759 101.3793103448276 M8.275862068965518 101.3793103448276 L20.689655172413794 101.3793103448276 M24.827586206896555 101.3793103448276 L28.965517241379313 101.3793103448276 M33.10344827586207 101.3793103448276 L41.37931034482759 101.3793103448276 M45.51724137931035 101.3793103448276 L49.65517241379311 101.3793103448276 M53.793103448275865 101.3793103448276 L66.20689655172414 101.3793103448276 M70.3448275862069 101.3793103448276 L74.48275862068967 101.3793103448276 M78.62068965517241 101.3793103448276 L107.58620689655173 101.3793103448276 M111.72413793103449 101.3793103448276 L115.86206896551725 101.3793103448276 M0 105.51724137931035 L4.137931034482759 105.51724137931035 M8.275862068965518 105.51724137931035 L20.689655172413794 105.51724137931035 M24.827586206896555 105.51724137931035 L28.965517241379313 105.51724137931035 M49.65517241379311 105.51724137931035 L57.931034482758626 105.51724137931035 M62.06896551724138 105.51724137931035 L66.20689655172414 105.51724137931035 M86.89655172413794 105.51724137931035 L91.0344827586207 105.51724137931035 M115.86206896551725 105.51724137931035 L120.00000000000001 105.51724137931035 M0 109.65517241379311 L4.137931034482759 109.65517241379311 M8.275862068965518 109.65517241379311 L20.689655172413794 109.65517241379311 M24.827586206896555 109.65517241379311 L28.965517241379313 109.65517241379311 M37.24137931034483 109.65517241379311 L41.37931034482759 109.65517241379311 M45.51724137931035 109.65517241379311 L49.65517241379311 109.65517241379311 M53.793103448275865 109.65517241379311 L82.75862068965517 109.65517241379311 M86.89655172413794 109.65517241379311 L91.0344827586207 109.65517241379311 M103.44827586206897 109.65517241379311 L120.00000000000001 109.65517241379311 M0 113.79310344827587 L4.137931034482759 113.79310344827587 M24.827586206896555 113.79310344827587 L28.965517241379313 113.79310344827587 M49.65517241379311 113.79310344827587 L57.931034482758626 113.79310344827587 M62.06896551724138 113.79310344827587 L78.62068965517241 113.79310344827587 M82.75862068965517 113.79310344827587 L86.89655172413794 113.79310344827587 M95.17241379310346 113.79310344827587 L99.31034482758622 113.79310344827587 M103.44827586206897 113.79310344827587 L107.58620689655173 113.79310344827587 M111.72413793103449 113.79310344827587 L120.00000000000001 113.79310344827587 M0 117.93103448275863 L28.965517241379313 117.93103448275863 M33.10344827586207 117.93103448275863 L49.65517241379311 117.93103448275863 M53.793103448275865 117.93103448275863 L57.931034482758626 117.93103448275863 M66.20689655172414 117.93103448275863 L74.48275862068967 117.93103448275863 M82.75862068965517 117.93103448275863 L86.89655172413794 117.93103448275863 M91.0344827586207 117.93103448275863 L103.44827586206897 117.93103448275863 M111.72413793103449 117.93103448275863 L115.86206896551725 117.93103448275863 "
                  fill={
                    Array [
                      0,
                      0,
                      0,
                      0,
                      1,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={
                    Array [
                      "stroke",
                      "strokeWidth",
                    ]
                  }
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={
                    Array [
                      0,
                      0.03529411764705882,
                      0.03529411764705882,
                      0.03529411764705882,
                      1,
                    ]
                  }
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth="4.137931034482759"
                  x={0}
                  y={0}
                />
              </RNSVGSvgView>
            </View>
          </View>
          <View />
          <RCTScrollView
            data={Array []}
            disableVirtualization={false}
            getItem={[Function]}
            getItemCount={[Function]}
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onContentSizeChange={[Function]}
            onEndReachedThreshold={2}
            onLayout={[Function]}
            onMomentumScrollEnd={[Function]}
            onScroll={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            renderItem={[Function]}
            scrollEventThrottle={50}
            stickyHeaderIndices={Array []}
            updateCellsBatchingPeriod={50}
            viewabilityConfigCallbackPairs={Array []}
            windowSize={21}
          >
            <View />
          </RCTScrollView>
        </View>
      </RCTScrollView>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
